File: benchmarks/execution/Python_Primitives/sort_m2_data1000_exec.py
from btypes.BInteger import BInteger
from sort_m2_data1000 import sort_m2_data1000


class sort_m2_data1000_exec():

    def __init__(self):
        self._sort_m2_data1000 = sort_m2_data1000()
        self.counter = BInteger(0)
        self.sorted = BInteger(0)

    def simulate(self):
        while (self.sorted.less(BInteger(500))).booleanValue():
            self.counter = BInteger(0)
            while (self.counter.less(BInteger(999).minus(BInteger(2).multiply(self.sorted)))).booleanValue():
                self._sort_m2_data1000.prog2()
                self.counter = self.counter.plus(BInteger(1))
            self.counter = BInteger(0)
            while (self.counter.less(self.sorted)).booleanValue():
                self._sort_m2_data1000.prog1()
                self.counter = self.counter.plus(BInteger(1))
            self._sort_m2_data1000.progress()
            self.sorted = self.sorted.plus(BInteger(1))
        while (self.sorted.less(BInteger(999))).booleanValue():
            self.counter = BInteger(0)
            while (self.counter.less(BInteger(999).minus(self.sorted))).booleanValue():
                self._sort_m2_data1000.prog1()
                self.counter = self.counter.plus(BInteger(1))
            self._sort_m2_data1000.progress()
            self.sorted = self.sorted.plus(BInteger(1))
        self._sort_m2_data1000.final_evt()

if __name__ == '__main__':
    execute = sort_m2_data1000_exec()
    execute.simulate()

File: benchmarks/execution/C++_Primitives/sort_m2_data1000_exec.cpp
#include <iostream>
#include <string>
#include <btypes_primitives/BUtils.hpp>
#include <btypes_primitives/StateNotReachableError.hpp>
#include <btypes_primitives/PreconditionOrAssertionViolation.hpp>
#include <btypes_primitives/BBoolean.hpp>
#include <btypes_primitives/BInteger.hpp>
#include "sort_m2_data1000.hpp"

#ifndef sort_m2_data1000_exec_H
#define sort_m2_data1000_exec_H

using namespace std;

class sort_m2_data1000_exec {

    public:



    private:


        sort_m2_data1000 _sort_m2_data1000;


        BInteger counter;
        BInteger sorted;
    public:
        sort_m2_data1000_exec() {
            counter = (BInteger(0));
            sorted = (BInteger(0));
        }

        void simulate() {
            while((sorted.less((BInteger(500)))).booleanValue()) {
                counter = (BInteger(0));
                while((counter.less((BInteger(999)).minus((BInteger(2)).multiply(sorted)))).booleanValue()) {
                    this->_sort_m2_data1000.prog2();
                    counter = counter.plus((BInteger(1)));
                }
                counter = (BInteger(0));
                while((counter.less(sorted)).booleanValue()) {
                    this->_sort_m2_data1000.prog1();
                    counter = counter.plus((BInteger(1)));
                }
                this->_sort_m2_data1000.progress();
                sorted = sorted.plus((BInteger(1)));
            }
            while((sorted.less((BInteger(999)))).booleanValue()) {
                counter = (BInteger(0));
                while((counter.less((BInteger(999)).minus(sorted))).booleanValue()) {
                    this->_sort_m2_data1000.prog1();
                    counter = counter.plus((BInteger(1)));
                }
                this->_sort_m2_data1000.progress();
                sorted = sorted.plus((BInteger(1)));
            }
            this->_sort_m2_data1000.final_evt();
        }

        BInteger _get_counter() const {
            return counter;
        }

        BInteger _get_sorted() const {
            return sorted;
        }

        friend std::ostream& operator<<(std::ostream &strm, const sort_m2_data1000_exec &machine) {
          strm << "_sort_m2_data1000: " << machine._sort_m2_data1000 << "\n";
          strm << "_get_counter: " << machine._get_counter() << "\n";
          strm << "_get_sorted: " << machine._get_sorted() << "\n";
          return strm;
        }

};

int main() {
    sort_m2_data1000_exec exec;
    exec.simulate();
    cout << exec << "\n";
    return 0;
}

#endif